CMakeLists.txt

```diff
@@ -12,6 +12,17 @@ set(CMAKE_COLOR_MAKEFILE ON)
 
 set(CMAKE_MACOSX_RPATH 1)
 
+# this is recommended for profiling using clion here:
+# https://www.jetbrains.com/help/clion/cpu-profiler.html#UsingTheProfiler
+# set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
+# set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0")
+# add_compile_options(-fno-omit-frame-pointer -mno-omit-leaf-frame-pointer)
+
+#to ignore asserts in profiling (supposed to work with g++):
+#https://stackoverflow.com/questions/5354314/how-can-i-completely-disable-calls-to-assert#
+add_compile_options(-DNDEBUG -g)
```

Comment:
We do not want to set these for all compilations. This should by default be commented out at least.

Alternatively, we could invest in creating some kind of configuration file which would determine these kinds of compilation flags and easier modification of behaviour of Mata and its compilation. Whether to turn off debug asserts, whether to preallocate underlying structures after the first insertion and how much space to preallocate, etc. I think this configuration file could prove beneficial.

Reply:
Ok, I will remove that, but teach me what to do in Clion :).